app_dart/lib/src/service/github_service.dart

```diff
@@ -15,27 +15,40 @@ class GithubService {
   final GitHub github;
 
   /// Lists the commits of the provided repository [slug] and [branch].
-  Future<List<RepositoryCommit>> listCommits(
-      RepositorySlug slug, String branch, int hours, Config config) async {
-    final Duration time = Duration(hours: hours);
+  Future<List<RepositoryCommit>> listCommits(RepositorySlug slug, String branch,
+      int lastCommitTimestampMills, Config config) async {
     ArgumentError.checkNotNull(slug);
     final PaginationHelper paginationHelper = PaginationHelper(github);
 
+    /// Return only one page when this is a new branch. Otherwise it will
+    /// return all commits prior to this release branch commit, leading to
+    /// heavy workload.
+    int pages;
+    if (lastCommitTimestampMills == 0) {
+      pages = 1;
+    }
+
     List<Map<String, dynamic>> commits = <Map<String, dynamic>>[];
+
+    /// [lastCommitTimestamp+1] excludes last commit itself.
     await for (Response response in paginationHelper.fetchStreamed(
```

Comment:
Nit: comment the url to the github api: https://developer.github.com/v3/repos/commits/#list-commits

Reply:
added.